src/components/scenes/EditTokenScene.tsx

```diff
@@ -175,14 +202,26 @@ function EditTokenSceneComponent(props: Props) {
           value={displayName}
           onChangeText={setDisplayName}
         />
-        <FilledTextInput
-          around={0.5}
-          autoCorrect={false}
-          autoFocus={false}
-          placeholder={lstrings.addtoken_contract_address_input_text}
-          value={contractAddress}
-          onChangeText={setContractAddress}
-        />
+        {customTokenTemplate.map(item => {
+          if (item.type === 'nativeAmount') return null
+          return (
+            <FilledTextInput
+              key={item.key}
+              around={0.5}
+              autoCorrect={false}
+              autoFocus={false}
+              placeholder={translateDescription(item.displayName)}
+              value={location.get(item.key) ?? ''}
+              onChangeText={value =>
+                setLocation(location => {
+                  const out = new Map(location)
+                  out.set(item.key, value)
+                  return out
+                })
+              }
```

Comment:
Move this to `handleChangeText(type: T)` handler with some input validation (check to make sure string is number, etc)? Where `T` is `EdgeObjectTemplate['type']`.